pkg/server/domain/service/webhook.go

```diff
@@ -187,28 +188,71 @@ func (c *webhookServiceImpl) patchComponentProperties(ctx context.Context, compo
 }
 
 func (c *customHandlerImpl) handle(ctx context.Context, webhookTrigger *model.ApplicationTrigger, app *model.Application) (interface{}, error) {
-	for comp, properties := range c.req.Upgrade {
-		component := &model.ApplicationComponent{
+	if c.req.Action == "" {
+		for comp, properties := range c.req.Upgrade {
+			component := &model.ApplicationComponent{
+				AppPrimaryKey: webhookTrigger.AppPrimaryKey,
+				Name:          comp,
+			}
+			if err := c.w.Store.Get(ctx, component); err != nil {
+				if errors.Is(err, datastore.ErrRecordNotExist) {
+					return nil, bcode.ErrApplicationComponentNotExist
+				}
+				return nil, err
+			}
+			if err := c.w.patchComponentProperties(ctx, component, properties.RawExtension()); err != nil {
+				return nil, err
+			}
+		}
+		return c.w.ApplicationService.Deploy(ctx, app, apisv1.ApplicationDeployRequest{
+			WorkflowName: webhookTrigger.WorkflowName,
+			Note:         "triggered by webhook custom",
+			TriggerType:  apisv1.TriggerTypeWebhook,
+			Force:        true,
+			CodeInfo:     c.req.CodeInfo,
+		})
+	} else {
+		workflow := &model.Workflow{
 			AppPrimaryKey: webhookTrigger.AppPrimaryKey,
-			Name:          comp,
+			Name:          webhookTrigger.WorkflowName,
 		}
-		if err := c.w.Store.Get(ctx, component); err != nil {
-			if errors.Is(err, datastore.ErrRecordNotExist) {
-				return nil, bcode.ErrApplicationComponentNotExist
+		c.w.Store.Get(ctx, workflow)
+		switch c.req.Action {
+		case "approve":
+			if err := c.w.WorkflowService.ResumeRecord(ctx, app, workflow, "", c.req.Step); err != nil {
+				return nil, err
 			}
-			return nil, err
-		}
-		if err := c.w.patchComponentProperties(ctx, component, properties.RawExtension()); err != nil {
-			return nil, err
+			return "workflow resumed successfully", nil
```

Comment:
The result can not be a string. Let's define a common struct.

Reply:
So, I think the result can be similar to the result of `Deploy` or `RollbackRecord`.

Reply:
Since the return struct of `ApplicationService.Deploy` and `ApplicationService.RollbackRecord` are of different type, Can you pls suggest a best suitable struct for the result.